vg/vggio/vggio.go

```diff
@@ -0,0 +1,461 @@
+	var (
+		ops    = c.gtx.Ops
+		dpi    = c.DPI()
+		min    = rect.Min
+		xmin   = min.X.Dots(dpi)
+		ymin   = min.Y.Dots(dpi)
+		rsz    = rect.Size()
+		width  = rsz.X.Dots(dpi)
+		height = rsz.Y.Dots(dpi)
+		dx     = float64(img.Bounds().Dx())
+		dy     = float64(img.Bounds().Dy())
+	)
+
+	c.ctx.push()
+	c.ctx.scale(1, -1)
+	c.ctx.translate(xmin, -ymin-height)
+	c.ctx.scale(width/dx, height/dy)
+	paint.NewImageOp(img).Add(ops)
+	paint.PaintOp{}.Add(ops)
+	c.ctx.pop()
+}
+
+var (
+	dbfonts = make(map[string][]text.FontFace)
+)
+
+func init() {
+	registerFont(text.Font{}, "Courier", "LiberationMono-Regular")
+	registerFont(
+		text.Font{Weight: text.Bold},
+		"Courier-Bold", "LiberationMono-Bold",
+	)
+	registerFont(
+		text.Font{Style: text.Italic},
+		"Courier-Oblique", "LiberationMono-Italic",
+	)
+	registerFont(
+		text.Font{
+			Style:  text.Italic,
+			Weight: text.Bold,
+		},
+		"Courier-BoldOblique", "LiberationMono-BoldItalic",
+	)
+
+	registerFont(text.Font{}, "Helvetica", "LiberationSans-Regular")
+	registerFont(
+		text.Font{Weight: text.Bold},
+		"Helvetica-Bold", "LiberationSans-Bold",
+	)
+	registerFont(
+		text.Font{Style: text.Italic},
+		"Helvetica-Oblique", "LiberationSans-Italic",
+	)
+	registerFont(
+		text.Font{
+			Style:  text.Italic,
+			Weight: text.Bold,
+		},
+		"Helvetica-BoldOblique", "LiberationSans-BoldItalic",
+	)
+
+	registerFont(text.Font{}, "Times-Roman", "LiberationSerif-Regular")
+	registerFont(
+		text.Font{Weight: text.Bold},
+		"Times-Bold", "LiberationSerif-Bold",
+	)
+	registerFont(
+		text.Font{Style: text.Italic},
+		"Times-Oblique", "LiberationSerif-Italic",
+	)
+	registerFont(
+		text.Font{
+			Style:  text.Italic,
+			Weight: text.Bold,
+		},
+		"Times-BoldOblique", "LiberationSerif-BoldItalic",
+	)
+}
+
+func collectionFor(fnt vg.Font) []text.FontFace {
+	name := collectionName(fnt.Name())
+	coll, ok := dbfonts[name]
+	if !ok {
+		return gofont.Collection()
+	}
+	return coll
+}
+
+func collectionName(name string) string {
+	// regroup fonts with name "Liberation-Italic", "Liberation-Bold", ...
+	// under the same collection "Liberation".
+	if strings.Contains(name, "-") {
+		i := strings.Index(name, "-")
+		name = name[:i]
+	}
+	return name
+}
+
+func registerFont(fnt text.Font, name, alias string) {
+	raw, err := vgfonts.Asset(alias + ".ttf")
+	if err != nil {
+		panic(fmt.Errorf("vggio: could not locate font %q: %+v", name, err))
+	}
+
+	face, err := opentype.Parse(raw)
+	if err != nil {
+		panic(fmt.Errorf("vggio: could not parse font %q: %+v", name, err))
+	}
+
+	fnt.Typeface = text.Typeface(name)
+	colName := collectionName(name)
+	dbfonts[colName] = append(dbfonts[colName], text.FontFace{
+		Font: fnt,
+		Face: face,
+	})
+
+	fnt.Typeface = text.Typeface(alias)
+	colName = collectionName(alias)
+	dbfonts[colName] = append(dbfonts[colName], text.FontFace{
+		Font: fnt,
+		Face: face,
+	})
+}
+
+var (
+	_ vg.Canvas      = (*Canvas)(nil)
+	_ vg.CanvasSizer = (*Canvas)(nil)
+)
```

Comment:
Can you place this at the top of the file? It's a statement of intent, so it should be seen early.